service/history/commandChecker.go

```diff
@@ -572,16 +585,46 @@ func (v *commandAttrValidator) validateStartChildExecutionAttributes(
 	}
 	attributes.TaskQueue = taskQueue
 
-	attributes.WorkflowExecutionTimeout = timestamp.DurationPtr(getWorkflowExecutionTimeout(targetNamespace,
-		timestamp.DurationValue(attributes.GetWorkflowExecutionTimeout()), v.config))
+	// TODO remove this if block after 1.5
+	if !v.config.EnableInfiniteTimeout() {
+		attributes.WorkflowExecutionTimeout = timestamp.DurationPtr(
+			common.GetWorkflowExecutionTimeout(
+				timestamp.DurationValue(attributes.GetWorkflowExecutionTimeout()),
+			),
+		)
+
+		attributes.WorkflowRunTimeout = timestamp.DurationPtr(
+			common.GetWorkflowRunTimeout(
+				timestamp.DurationValue(attributes.GetWorkflowRunTimeout()),
+				timestamp.DurationValue(attributes.GetWorkflowExecutionTimeout()),
+			),
+		)
+		// Inherit workflow task timeout from parent workflow execution if not provided on command
+
+		if timestamp.DurationValue(attributes.GetWorkflowTaskTimeout()) <= 0 {
+			attributes.WorkflowTaskTimeout = parentInfo.DefaultWorkflowTaskTimeout
+		}
+		return nil
+	}
 
-	attributes.WorkflowRunTimeout = timestamp.DurationPtr(getWorkflowRunTimeout(targetNamespace,
-		timestamp.DurationValue(attributes.GetWorkflowRunTimeout()), timestamp.DurationValue(attributes.GetWorkflowExecutionTimeout()), v.config))
+	// workflow execution timeout is left as is
+	//  if workflow execution timeout == 0 -> infinity
 
-	// Inherit workflow task timeout from parent workflow execution if not provided on command
-	if timestamp.DurationValue(attributes.GetWorkflowTaskTimeout()) <= 0 {
-		attributes.WorkflowTaskTimeout = parentInfo.DefaultWorkflowTaskTimeout
-	}
+	attributes.WorkflowRunTimeout = timestamp.DurationPtr(
+		common.OverrideWorkflowRunTimeout(
+			timestamp.DurationValue(attributes.GetWorkflowRunTimeout()),
+			timestamp.DurationValue(attributes.GetWorkflowExecutionTimeout()),
+		),
+	)
+
+	attributes.WorkflowTaskTimeout = timestamp.DurationPtr(
```

Comment:
Just to confirm, it is intentional to remove the logic workflow task timeout from parent?

Reply:
> Just to confirm, it is intentional to remove the logic workflow task timeout from parent?

Existing logic tries to special handling workflow task timeout not specified (not sure why), I believe we should unify all task timeout default value, rather than special handling (yet another special case)